src/backend/executor/execProcnode.c

```diff
@@ -1242,6 +1245,9 @@ MultiExecProcNode(PlanState *node)
 	}
 
 	TRACE_POSTGRESQL_EXECPROCNODE_EXIT(GpIdentity.segindex, currentSliceId, nodeTag(node), node->plan->plan_node_id);
+
+	if (query_info_collect_hook)
```

Comment:
I believe that `query_info_collect_hook` should be **always** inserted before `TRACE_POSTGRESQL_EXECPROCNODE_EXIT` macros. Though this macros defines as empty, some debug extensions can redefine it and expect it to be the last operation with the node (for time tracking for example).

Reply:
The same thing is actual for the master branch [draft](https://github.com/arenadata/gpdb/pull/207).

Reply:
For exactly the same purpose (to obtain "last operation" guarantee) we may wish to leave the hook in its current location. The macro cannot be redefined at runtime, thus its user would have to be aware of the fact that it is not the last operation in `ExecProcNode()`.

Taking into account that `gpperfmon` packets [are sent earlier](https://github.com/arenadata/gpdb/pull/208/files#diff-1255433ef357570f11a775d6a32fab02584024c0708986e0b16837717ee355b0R1161) than the macro is called, the change you propose makes sense.

Reply:
Changed in https://github.com/arenadata/gpdb/pull/208/commits/7b7ceb0b5927b4eefe2fe37e6a80e3ddab5ff898.

Reply:
> some debug extensions can redefine it

not debug extensions but static tracing tools (ebpf tools, as example)